source/PlayState.hx

```diff
@@ -19,177 +20,272 @@ class PlayState extends FlxState
 	public var GRAVITY(default, never):Float = 600;
 
 	private var map:FlxTilemap;
+	private var _map:FlxOgmoLoader;
+	private var _mGround:FlxTilemap;
 	public var player:Player;
 	private var flagpole:FlagPole;
 	private var platform:Platforms;
 	private var trap:Trap;
  	private var coins:FlxGroup;
-  private var flag_x_loc:Int = 17;
+	private var flag_x_loc:Int = 37;
 	private var flag_y_loc:Int = 11;
 
 	public static var hud:HeadsUpDisplay;
 
 	public var sprites:FlxTypedGroup<FlxObject> = new FlxTypedGroup<FlxObject>();
-
-	private var blockGroup:FlxTypedGroup<Block> = new FlxTypedGroup<Block>(10);
+	public var trapGroup:FlxTypedGroup<Trap> = new FlxTypedGroup<Trap>();
+	private var blockGroup:FlxTypedGroup<Block> = new FlxTypedGroup<Block>();
+	private var platformGroup:FlxTypedGroup<Platforms> = new FlxTypedGroup<Platforms>();
+	
 	
 	// Enemies
-	private var dtmEnemy1:DontTouchMe;
+	private var dtmEnemy:DontTouchMe;
+	private var dtmGroup:FlxTypedGroup<DontTouchMe> = new FlxTypedGroup<DontTouchMe>();
+	private var turEnemy:Turtle;
+	private var turGroup:FlxTypedGroup<Turtle> = new FlxTypedGroup<Turtle>();
+	private var flyingEnemy:FlyingTurtle;
+	private var flyingGroup:FlxTypedGroup<FlyingTurtle> = new FlxTypedGroup<FlyingTurtle>();
 	private var sentry1:Sentry;
 	private var bullets:FlxTypedGroup<Bullet>;
 	
-	/**
-	 * bulletHitPlayer
-	 * Logic for when a bullet overlaps with a player
-	 * 
-	 * @param	player	A player's character
-	 * @param	bullet	A bullet sprite
-	 */
-	public function bulletHitPlayer(player:Player, bullet:FlxObject):Void
-	{
-		if (!player.star) {
-			player.kill();
-		}
-		
-		bullet.kill();
-	}
-
-	
 	override public function create():Void
 	{
-		//if (hud == null){
-			hud = new HeadsUpDisplay(0, 0, "MARIO");
-		//}
 		super.create();
-    
-    		/*Create the flagpole at the end of the level 
-		 * This will also instantiate the flag
-		 * flag_x_loc is the number of blocks to the right where we want the flag
-		 * flag_y_loc is the number of blocks down we want the flag
-		*/
-		flagpole = new FlagPole(32*flag_x_loc, 32*flag_y_loc);
-		add(flagpole);
-		add(flagpole.flag);
-
-		player = new Player(50, 50);
-		add(player);
-
-		//Add player (and any other sprites) to group
-		sprites.add(player);
-		
-		//create new moving platform
-		platform =  new Platforms(250, 150, 3, 100, 100, 50, 50, player);
-		platform.immovable = platform.solid = true;
-		platform.allowCollisions = FlxObject.UP;
-		platform.inContact = false;
-		add(platform);
-
-		add(player.hitBoxComponents);
 		
-		//Coins are added to a group, coin group added to playstate
 		coins = new FlxGroup();
```

Comment:
I missed that comment, but the coins are added down with everything else. This line instantiates the coin group so the coins can be added to that group and then added to the game.